pkg/cloud/gcp/actuators/machine/machine_scope.go

```diff
@@ -18,12 +17,17 @@ import (
 
 // machineScopeParams defines the input parameters used to create a new MachineScope.
 type machineScopeParams struct {
-	coreClient controllerclient.Client
-	machine    *machinev1.Machine
+	context.Context
```

Comment:
Out of interest, why embed the context rather than have it as an explicit field?

Reply:
I don't think there any benefit in it, but we have the same in vsphere https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/machine_scope.go#L33 + I saw a similar thing in upstream https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/pkg/context/controller_manager_context.go#L35